pkg/deploy-manager/common.go

```diff
@@ -27,6 +27,35 @@ func (t *DeployManager) CreateNamespace(namespace string) error {
 	return nil
 }
 
+// DeleteStorageClusterAndWait deletes a storageClusterCR and waits on it to terminate
+func (t *DeployManager) DeleteStorageClusterAndWait(namespace string) error {
+	err := t.deleteStorageCluster()
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	timeout := 300 * time.Second
+	interval := 10 * time.Second
+
+	// Wait for storagecluster and cephCluster to terminate
+	err = utilwait.PollImmediate(interval, timeout, func() (done bool, err error) {
```

Comment:
Technically, CephCluster will/should be deleted before StorageCluster due to OwnerRef. So, it makes more sense to check for deletion of CephCluster first and then for StorageCluster.

Reply:
since its running inside polling function. I don't think the sequence matter. The idea was to just check for any remaining of cephCluster or StorageCluster. Anyway i moved cephCluster check before storageCluster.